astropy/constants/__init__.py

```diff
@@ -81,17 +118,23 @@ def set_enabled_constants(modname):
     # Ignore warnings about "Constant xxx already has a definition..."
     with warnings.catch_warnings():
         warnings.simplefilter('ignore')
-        _utils._set_c(codata, iaudata, module,
+        _utils._set_c(codata_context, iaudata_context, module,
                       not_in_module_only=False, set_class=True)
 
     try:
         yield
     finally:
         with warnings.catch_warnings():
             warnings.simplefilter('ignore')
-            # NOTE: Update this when default changes.
-            _utils._set_c(codata2014, iau2015, module,
+            _utils._set_c(codata, iaudata, module,
                           not_in_module_only=False, set_class=True)
+            del codata_context, iaudata_context
```

Comment:
Is this necessary?  Wouldn't it happen automatically when execution reaches the end of this?